react/src/components/ManageAppsModal.tsx

```diff
@@ -1,101 +1,210 @@
+import { useSuspendedBackendaiClient } from '../hooks';
 import BAIModal, { BAIModalProps } from './BAIModal';
 import { useWebComponentInfo } from './DefaultProviders';
 import Flex from './Flex';
 import { DeleteOutlined, PlusOutlined, CheckOutlined } from '@ant-design/icons';
-import { Input, Typography, Space, Button, Form } from 'antd';
-import React from 'react';
+import { Input, Button, Form, message, Typography } from 'antd';
+import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
 
-interface Props extends BAIModalProps {}
-const { Text } = Typography;
-const ManageAppsModal: React.FC<Props> = ({ ...baiModalProps }) => {
+const ManageAppsModal: React.FC<BAIModalProps> = ({ ...baiModalProps }) => {
+  const [validateDetail, setValidateDetail] = useState('');
   const { t } = useTranslation();
+  const [form] = Form.useForm();
   const { value, dispatchEvent } = useWebComponentInfo();
+  const baiClient = useSuspendedBackendaiClient();
+
   let parsedValue: {
-    open: boolean;
+    image: any;
     servicePorts: any;
   };
   try {
     parsedValue = JSON.parse(value || '');
   } catch (error) {
     parsedValue = {
-      open: false,
+      image: {},
       servicePorts: [],
     };
   }
-  const { open, servicePorts } = parsedValue;
-  const onFinish = (values: any) => {
-    console.log('Saved settings. ', values);
+  const { image, servicePorts } = parsedValue;
+
+  const handleOnclick = async () => {
+    try {
+      await form.validateFields();
+
+      const values = form
+        .getFieldValue('apps')
+        .map((item: { app: string; protocol: string; port: string }) => {
+          return `${item.app}:${item.protocol}:${item.port}`;
+        })
+        .join(',');
+
+      const { result } = await baiClient.image.modifyLabel(
+        image.registry,
+        image.name,
+        image.tag,
+        'service-ports',
+        values,
+      );
+
+      if (result === 'ok') {
+        message.success(t('environment.DescServicePortModified'));
+        dispatchEvent('ok', null);
+        return;
+      }
+      message.error(t('dialog.ErrorOccurred'));
+    } catch (info: any) {
+      setValidateDetail(info.errorFields[0].errors[0]);
+      return;
+    }
   };
 
   return (
     <BAIModal
-      open={open}
+      destroyOnClose
+      open={true}
       onCancel={() => {
         dispatchEvent('cancel', null);
       }}
       centered
       title={t('environment.ManageApps')}
       {...baiModalProps}
-      footer={[
-        <Flex direction="row" justify="between">
-          <Button type="text" danger>
-            {t('button.Reset')}
-          </Button>
-          <Button type="primary" icon={<CheckOutlined />}>
-            {t('button.Save')}
-          </Button>
-        </Flex>,
-      ]}
+      footer={
```

Comment:
You can implement without using a footer if there are only two buttons. Please change them to "Cancel" and "Save".

Reply:
Since the footer shows the validation result of each field, so only the buttons in the footer used antd's default modal buttons.